pkgs/development/tools/build-managers/bazel/bazel_git/default.nix

```diff
@@ -0,0 +1,624 @@
+      # "error: no visible @interface for 'NSDictionary' declares the selector
+      # 'initWithContentsOfURL:error:'"
+      # This can be removed when the apple_sdk is upgraded beyond 10.13+
+      sed -i '/initWithContentsOfURL:versionPlistUrl/ {
+        N
+        s/error:nil\];/\];/
+      }' tools/osx/xcode_locator.m
+
+      # append the PATH with defaultShellPath in tools/bash/runfiles/runfiles.bash
+      echo "PATH=\$PATH:${defaultShellPath}" >> runfiles.bash.tmp
+      cat tools/bash/runfiles/runfiles.bash >> runfiles.bash.tmp
+      mv runfiles.bash.tmp tools/bash/runfiles/runfiles.bash
+
+      patchShebangs .
+    '';
+    in lib.optionalString stdenv.hostPlatform.isDarwin darwinPatches
+     + genericPatches;
+
+  buildInputs = [
+    buildJdk
+    python3
+  ];
+
+  # when a command can’t be found in a bazel build, you might also
+  # need to add it to `defaultShellPath`.
+  nativeBuildInputs = [
+    installShellFiles
+    zip
+    python3
+    unzip
+    makeWrapper
+    which
+    customBash
+  ] ++ lib.optionals (stdenv.isDarwin) [ cctools libcxx CoreFoundation CoreServices Foundation ];
+
+  # Bazel makes extensive use of symlinks in the WORKSPACE.
+  # This causes problems with infinite symlinks if the build output is in the same location as the
+  # Bazel WORKSPACE. This is why before executing the build, the source code is moved into a
+  # subdirectory.
+  # Failing to do this causes "infinite symlink expansion detected"
+  preBuildPhases = ["preBuildPhase"];
+  preBuildPhase = ''
+    mkdir bazel_src
+    shopt -s dotglob extglob
+    mv !(bazel_src) bazel_src
+  '';
+  # Needed to build fish completion
+  propagatedBuildInputs = [ python3.pkgs.absl-py ];
+  buildPhase = ''
+    # Increasing memory during compilation might be necessary.
+    # export BAZEL_JAVAC_OPTS="-J-Xmx2g -J-Xms200m"
+
+    # Since we're building bazel from git sources, we need to
+    # provide protoc needed to generate some bootstrap sources
+    mkdir -p ./bazel_src/derived/jars
+    cp ${protobuf-java}/protobuf-{java,java-util}-[0-9]*.jar ./bazel_src/derived/jars/
+    export PROTOC="${protobuf}/bin/protoc"
+    ${buildPackages.stdenv.cc}/bin/g++ -I "${protobuf}/include" -L "${protobuf}/lib" -lpthread -lm -lprotoc -lprotobuf bazel_src/third_party/grpc/compiler/src/java_plugin/cpp/{java_generator.h,java_generator.cpp,java_plugin.cpp} -o grpc-java-plugin
+    export GRPC_JAVA_PLUGIN="$(pwd)"/grpc-java-plugin
+
+    # Release bazel uses "version-date (short_rev)" format.
+    # What we build from git isn't exactly a release version.
+    # So let's produce label similar but not in identical format.
+    # Note that our .bazelversion is 3.9999 as determined by
+    # bazel executable name.
+    local version="$(grep -m 1 '## Release' bazel_src/CHANGELOG.md \
+                       | sed -E 's|.*Release (.*) \(.*\)|\1|')"
+    export EMBED_LABEL="${version} ~=$version (@${commit})"
+    ${customBash}/bin/bash ./bazel_src/compile.sh
```

Comment:
also fails in sandbox
```
Building Bazel from scratch......
Building Bazel with Bazel.
...
ERROR: no such package '@bazel_toolchains//rules': java.io.IOException: Error downloading [https://mirror.bazel.build/github.com/bazelbuild/bazel-toolchains/releases/download/3.1.0/bazel-toolchains-3.1.0.tar.gz, https://github.com/bazelbuild/bazel-toolchains/releases/download/3.1.0/bazel-toolchains-3.1.0.tar.gz] to /build/bazel_AFAZt0Fc/out/external/bazel_toolchains/temp4351923723686256938/bazel-toolchains-3.1.0.tar.gz: Unknown host: github.com
```
probably 2nd phase doesn't use our distdir bazelrc setting because bazel-toolchains-3.1.0.tar.gz is present in src-deps.json